runtime/runtime_test.go

```diff
@@ -7979,8 +7994,141 @@ func TestRuntimeInvalidatedResourceUse(t *testing.T) {
+                access(all) var parent: &OuterResource?
+                access(all) var vault: @VictimContract.Vault?
+
+                init(_ name: String) {
+                    self.name = name
+                    self.parent = nil
+                    self.vault <- nil
+                }
+
+                access(all) fun setParent(_ parent: &OuterResource) {
+                    self.parent = parent
+                }
+
+                access(all) fun setVault(_ vault: @VictimContract.Vault) {
+                    self.vault <-! vault
+                }
+
+                destroy() {
+                    self.parent!.shenanigans()
+                    var vault: @VictimContract.Vault <- self.vault!
+                    self.parent!.collect(<- vault)
+                }
+            }
+
+            access(all) resource OuterResource {
+                access(all) var inner1: @InnerResource
+                access(all) var inner2: @InnerResource
+                access(all) var collector: &VictimContract.Vault
+
+                init(_ victim: @VictimContract.Vault, _ collector: &VictimContract.Vault) {
+                    self.collector = collector
+                    var i1 <- create InnerResource("inner1")
+                    var i2 <- create InnerResource("inner2")
+                    self.inner1 <- i1
+                    self.inner2 <- i2
+                    self.inner1.setVault(<- victim)
+                    self.inner1.setParent(&self as &OuterResource)
+                    self.inner2.setParent(&self as &OuterResource)
+                }
+
+                access(all) fun shenanigans() {
+                    self.inner1 <-> self.inner2
+                }
+
+                access(all) fun collect(_ from: @VictimContract.Vault) {
+                    self.collector.deposit(from: <- from)
+                }
+            }
+
+            access(all) fun doubleBalanceOfVault(_ vault: @VictimContract.Vault): @VictimContract.Vault {
+                var collector <- vault.withdraw(amount: 0.0)
+                var outer <- create OuterResource(<- vault, &collector as &VictimContract.Vault)
+                destroy outer
+                return <- collector
+            }
+
+            access(all) fun attack() {
+                var v1 <- VictimContract.faucet()
+                var v2 <- AttackerContract.doubleBalanceOfVault(<- v1)
+                destroy v2
+           }
+        }`,
+		signerAccount.HexWithPrefix(),
+	))
+
+	// Deploy Attacker
+
+	deployAttacker := DeploymentTransaction("AttackerContract", attacker)
+
+	err := runtime.ExecuteTransaction(
+		Script{
+			Source: deployAttacker,
+		},
+		Context{
+			Interface: runtimeInterface,
+			Location:  nextTransactionLocation(),
+		},
+	)
+	RequireError(t, err)
+
+	assertRuntimeErrorIsUserError(t, err)
+
+	var parserError parser.Error
+	require.ErrorAs(t, err, &parserError)
 
+	assert.IsType(t, &parser.CustomDestructorError{}, parserError.Errors[0])
```

Comment:
This and the below ones are no longer valid, since destructors are no longer supported and those edge cases were only possible to be reached in the destructor. So updated the test to assert the new static error (`CustomDestructorError`).

We could technically move these tests to checker tests, but in future, if/when custom destructors are allowed, these would become valid tests again, so keeping it here would make life easier + won't miss out on adding these back again.